lib/Sema/CSDiagnostics.cpp

```diff
@@ -3428,6 +3431,75 @@ bool MissingMemberFailure::diagnoseInLiteralCollectionContext() const {
   return false;
 }
 
+bool MissingMemberFailure::diagnoseForSubscriptMemberWithTupleBase() const {
+  auto locator = getLocator();
+  auto baseType = resolveType(getBaseType())->getWithoutSpecifierType();
+
+  if (!locator->isLastElement<LocatorPathElt::SubscriptMember>())
```

Comment:
Nit: For clarity you could remove this check and replace `castToExpr<SubscriptExpr>` with `getAsExpr<SubscriptExpr>` + `if (!SE) return false;`

Reply:
Fixed!